### PR TITLE
LPS-192690 KB schedule modal

### PR DESCRIPTION
**Pull request by ambrinchaudhary**
Continued work from https://github.com/liferay-lima/liferay-portal/pull/4632. 

Hi @sofiamendozalr, I added some work on your pull before even review it, so you can continue working on it while I am on PTO: 

- I removed the first two commits, were unneccessary. 
- Now the modal include the footer buttons, so we han handle them internally in terms of disabling, adding more buttons...Please check how are they connected to the opener, and the events involved.  
- It is now working, I save the date in a hidden input (please check with @AliciaGarciaGarcia the name, maybe she expects another name - then please change it) -> Edited, I have checked with her and it should be renamed to "displayDate". 
- the problem you have with the `currentDate` inside the useEffect is now solved, I think that state variable was not necessary. 

Apart from the changes I request, please review these: 

- The DatePicker closes when I select a date, but I didn't have the change to set the time and it returns an error. Could we avoid this "closing"? 
- It fails when the time is 00:00, even if the date is after today

### CODE REVIEW

**Comment by liferay-lima** [comment]
ci:test:sf

**Comment by liferay-lima** [comment]
ci:test:relevant

File: modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/js/EditKBArticle.js [hunk 1]
```diff
@@ -109,17 +148,35 @@ export default function EditKBArticle({kbArticle, namespace, publishAction}) {
...
-			contextualSidebarButtonOnClick
-		),
-		attachListener(form, 'submit', () => {
-			beforeSubmit();
-		}),
-	];
+	let eventHandlers = null;
+
+	if (Liferay.FeatureFlags['LPS-188060']) {
+		eventHandlers = [
```
Comment (ambrinchaudhary) [inline modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/js/EditKBArticle.js:154]: You cam simplify this.
```
eventHandlers = [
....
]

if (Liferay.FeatureFlags['LPS-188060']) {
   eventHandlers.push(...) (or .`add,` I never remember the fn)
}
```

File: modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/js/components/ScheduleModal.js [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+		getOpener().Liferay.fire(SCHEDULE_EVENT_NAME);
+		closeModal();
+	};
+
+	useEffect(() => {
+		if (
+			isAfter(Date.parse(scheduledDate), Date.now()) ||
+			(Number.isNaN(Date.parse(scheduledDate)) && !scheduledDate)
+		) {
+			setInvalidDate(false);
```
Comment (ambrinchaudhary) [inline modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/js/components/ScheduleModal.js:45]: Remove if/else and simplify: 
```
setInvalidDate(isAfter(Date.parse(scheduledDate), Date.now()) ||
			(Number.isNaN(Date.parse(scheduledDate)) && !scheduledDate)
```

File: modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/js/components/ScheduleModal.js [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+	}, [scheduledDate]);
+
+	return (
+		<div className="schedule-modal">
+			<div className="container-fluid p-4">
+				<p className="text-secondary">
+					{Liferay.Language.get('set-date-and-time-for-publication')}
+				</p>
+
+				{invalidDate === true ? (
```
Comment (ambrinchaudhary) [inline modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/js/components/ScheduleModal.js:59]: All this section can be simplified so there is no need to have two different ClayDatePickers to maintain: 
```
<div className={classnames({'has-error': invalidDate})}>  --> use the className utility
<ClayDatePicker .../>
{invalidDate && <ClayAlert className="mt-1" .../>}
```

**Comment by WYZ-Issac** [comment]
ci:test:relevant

**Comment by adolfopa** [comment]
See https://github.com/liferay-echo/liferay-portal/pull/13526

**Pull request closed without merging**